src/components/notebook-utils.js

```diff
@@ -58,6 +67,9 @@ export const notebookNameInput = ({ inputProps, ...props }) => h(ValidatedInput,
   }
 })
 
+export const getAnalysisType = name => name.endsWith('.ipynb') ? 'python' : 'r'
```

Comment:
I miss scala case classes here :(. Not sure if there's a cleaner way to map from type to the various use cases (i.e. the file extension associated with an analysisType)

Reply:
Yeah, you'd do a switchcase on the last chunk of the name. I'd encourage you to use the same method of selection between this and `getAnalysisFileExtension`, whichever way you do it.